Sources/Core/GTMSessionFetcher.m

```diff
@@ -2013,14 +2069,28 @@ - (void)forgetSessionIdentifierForFetcherWithoutSyncCheck {
 
 // External stop method
 - (void)stopFetching {
+  BOOL triggerCallback;
   @synchronized(self) {
     GTMSessionMonitorSynchronized(self);
 
     // Prevent enqueued callbacks from executing. The completion handler will still execute if
     // the property `stopFetchingTriggersCompletionHandler` is `YES`.
     _userStoppedFetching = YES;
+
+    // `stopFetchReleasingCallbacks:` will dequeue it if there is a sevice throttled
+    // delay, so the canceled callback needs to be directly triggered since the serivce
+    // won't attempt to restart it.
+    triggerCallback = _delayState == kDelayStateServiceDelayed && self.stopFetchingTriggersCompletionHandler;
   }  // @synchronized(self)
-  [self stopFetchReleasingCallbacks:!self.stopFetchingTriggersCompletionHandler];
+
+  if (triggerCallback) {
+    NSError *error = [NSError errorWithDomain:kGTMSessionFetcherErrorDomain
+                                         code:GTMSessionFetcherErrorUserCancelled
+                                     userInfo:nil];
```

Comment:
Clients can see it if they enable the callbacks for cancel.

I'm going to hold off on adding it, right now most of the errors returned don't set that key and the fetcher is also usually right there in the callback, so developers should already have it. But it might be worth fixing across the board in a different PR.